dvc/main.py

```diff
@@ -53,6 +54,15 @@ def main(argv=None):
         ret = 253
     except DvcParserError:
         ret = 254
+    except UnicodeError:
+        if is_py2:
+            logger.exception(
+                "unicode is not fully supported in DVC for Python 2.\n"
+                "Python 2.7 will not be maintained past 2020.\n"
+                "Please upgrade to Python 3 to work correctly with unicode.\n"
+                "Exception"
+            )
+        ret = 255
     except Exception:  # pylint: disable=broad-except
         logger.exception("unexpected error")
```

Comment:
How about we do it like so:
```
except Exception as exc:  # pylint: disable=broad-except
    if type(exc) == UnicodeError:
        logger.exception(
            "unicode is not supported in DVC for Python 2 "
            "(end-of-life January 1, 2020), please upgrade to Python 3"
        )
    else:
        logger.exception("unexpected error")
    ret = 255

Reply:
what's the benefit from doing this instead of catching `UnicodeError` before?

Reply:
@mroutis It is broken in the current implementation https://github.com/iterative/dvc/pull/2538/files#r329275279 .

Reply:
It actually looks weird that we are checking `if isinstance(exc, UnicodeError)` when `except UnicodeError` does this.

I'm definitely overlooking something :see